File: javascript/src/televentas/queja.js
export class Queja {
  /**
   * @param {number} idQueja
   * @param {Date} fecha
   * @param {string} motivo
   * @param {string} descripcion
   * @param {import('./ordenCompra.js').OrdenCompra | null} orden
   */
  constructor(idQueja, fecha, motivo, descripcion, orden) {
    this.idQueja = idQueja;
    this.fecha = fecha;
    this.motivo = motivo;
    this.descripcion = descripcion;
    /** @type {import('./ordenCompra.js').OrdenCompra | null} */
    this.orden = orden ?? null;
    this.estado = "registrada";
  }

  getIdQueja() {
    return this.idQueja;
  }

  getFecha() {
    return this.fecha;
  }

  getMotivo() {
    return this.motivo;
  }

  getDescripcion() {
    return this.descripcion;
  }

  getEstado() {
    return this.estado;
  }

  /** @returns {import('./ordenCompra.js').OrdenCompra | null} */
  getOrden() {
    return this.orden;
  }

  setEstado(estado) {
    this.estado = estado;
  }

  registrar() {
    this.estado = "registrada";
  }

  remitir() {
    this.estado = "remitida";
  }

  cerrar() {
    this.estado = "cerrada";
  }
}

File: java/src/televentas/Queja.java
package televentas;

import java.time.LocalDate;
import java.util.Optional;

public class Queja {

    private final int idQueja;
    private final LocalDate fecha;
    private final String motivo;
    private final String descripcion;
    private final Optional<OrdenCompra> orden;
    String estado;

    public Queja(int idQueja, LocalDate fecha, String motivo, String descripcion, OrdenCompra orden) {
        this.idQueja = idQueja;
        this.fecha = fecha;
        this.motivo = motivo;
        this.descripcion = descripcion;
        this.orden = Optional.ofNullable(orden);
        this.estado = "registrada";
    }

    public int getIdQueja() {
        return idQueja;
    }

    public LocalDate getFecha() {
        return fecha;
    }

    public String getMotivo() {
        return motivo;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public String getEstado() {
        return estado;
    }

    public Optional<OrdenCompra> getOrden() {
        return orden;
    }

    public void setEstado(String estado) {
        this.estado = estado;
    }

    public void registrar() {
        this.estado = "registrada";
    }

    public void remitir() {
        this.estado = "remitida";
    }

    public void cerrar() {
        this.estado = "cerrada";
    }
}
